swig/python/setup.py.in

```diff
@@ -47,7 +47,7 @@ class CMakeBuild(build_ext):
             '-DSWIG_IMPORT=ON',
             '-DWITH_PYTHON=ON',
             '-DWITH_EXAMPLES=OFF',
-            '-DPYTHON_INCLUDE_DIR=' + self.include_dirs[0],
+            '-DPYTHON_INCLUDE_DIR=' + self.include_dirs[-1],
```

Comment:
Hi @andiradulescu, we've been hitting this on pyodide/pyodide#4936 too and I think this should point to the include directory of the Python interpreter being used through `sysconfig` (this requires an `import sysconfig` statement too that I can't suggest through GitHub code suggestions):

```suggestion
            '-DPYTHON_INCLUDE_DIR=' + sysconfig.get_path('include'),
```

Reply:
Thanks! I will apply these changes. I got the same suggestion [here](https://github.com/astral-sh/uv/issues/5004#issuecomment-2226122684).